package/src/ssr/NextSSRInMemoryCache.tsx

```diff
@@ -18,21 +16,6 @@ export class NextSSRInMemoryCache extends InMemoryCache {
   };
   constructor(config?: InMemoryCacheConfig) {
     super(config);
-
-    this.registerWindowHook();
-  }
-  private registerWindowHook() {
-    if (typeof window !== "undefined") {
-      if (Array.isArray(window[ApolloResultCache] || [])) {
-        registerLateInitializingQueue(ApolloResultCache, (data) =>
-          this.write(data)
-        );
-      } else {
-        throw new Error(
-          "On the client side, only one instance of `NextSSRInMemoryCache` can be created!"
-        );
-      }
-    }
```

Comment:
Huh, good question. If a real `watchQuery/fetchQuery` call latches onto our fake query before it finishes, it wouldn't be necessary - but if it doesn't, it would be necessary.
So maybe we could create the fake ongoing query, and then immediately call `QueryManager.fetchQuery` to make sure its' results gets written into the store? Or we just call `write` after resolving the query? (The latter might have a lower risk of race conditions 🤔 )

Reply:
Sounds good :) Addressed in [9a74dc6](https://github.com/apollographql/apollo-client-nextjs/pull/38/commits/9a74dc607f20aa1239cfdeee31f0e0eb157b5e43).